### ISSUE TITLE
[Swarm Mode] Distributed requests are slow and seem to not most of the time

### ISSUE THREAD

**Issue by mackeyja92** [vfarcic/docker-flow-proxy#34]
Using `distributed=true` in my curl command it seems very slow to respond/just doesn't respond at all. It seems to be doing the work sometimes but frequently it never updates consul or changes any configs. In the example below I am doing a remove but it is also the same for reconfigure. Also, I know that all the nodes are up and responding just fine because I can curl all of them individually on `:80/` and get the 503.

```
curl -i -X GET -v "127.0.0.1:8080/v1/docker-flow-proxy/remove?serviceName=nginx1&distribute=true"
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> GET /v1/docker-flow-proxy/remove?serviceName=nginx1&distribute=true HTTP/1.1
> User-Agent: curl/7.35.0
> Host: 127.0.0.1:8080
> Accept: */*
>
< HTTP/1.1 500 Internal Server Error
HTTP/1.1 500 Internal Server Error
< Date: Mon, 22 Aug 2016 18:05:16 GMT
Date: Mon, 22 Aug 2016 18:05:16 GMT
< Content-Length: 304
Content-Length: 304
< Content-Type: text/plain; charset=utf-8
Content-Type: text/plain; charset=utf-8

<
* Connection #0 to host 127.0.0.1 left intact
{"Status":"NOK","Message":"Could not send distribute request to the following addresses: [172.20.1.9]","ServiceName":"nginx1","ServiceColor":"","ServicePath":null,"ServiceDomain":"","ConsulTemplateFePath":"","ConsulTemplateBePath":"","PathType":"","SkipCheck":false,"Mode":"","Port":"","Distribute":true}
```

Another example of it not even being able to connect to port `8080`.

```
curl -i -X GET -v "127.0.0.1:8080/v1/docker-flow-proxy/remove?serviceName=nginx1&distribute=true"
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
^C
```

Here is the useful part of my service definition.

```
"Name": "sproxy",
"TaskTemplate": {
  "ContainerSpec": {
    "Image": "vfarcic/docker-flow-proxy",
    "Env": [
      "MODE=swarm",
      "CONSUL_ADDRESS=consul:8500",
      "SERVICE_NAME=sproxy",
      "PROXY_INSTANCE_NAME=sproxy"
    ]
},
```

**Comment by mackeyja92** [comment]
These might be related to the issues in docker that have been fixed in 1.12.1. Let me update and test again.

**Comment by vfarcic** [comment]
I had a lot of problems with 1.12, especially with networking. Those that I experienced were fixed in 1.12.1. Please let me know if it works with the latest version.

**Comment by mackeyja92** [comment]
I am going with I think this is a docker issue and not a flow proxy issue. Sorry about that!
